app/controllers/documents/document_upload_question_controller.rb

```diff
@@ -9,30 +9,38 @@ def edit
       return if self.class.document_type.nil?
 
       @documents = current_intake.documents.of_type(document_type)
-      @form = form_class.new(document_type, current_intake, form_params)
+      @form ||= form_class.new(document_type, current_intake, form_params)
     end
 
     def update
       return if self.class.document_type.nil?
 
       @form = form_class.new(document_type, current_intake, form_params)
       if @form.valid?
-        form_saved = @form.save
+        @form.save
         after_update_success
         track_document_upload
+        if @form.try(:next_step)
+          redirect_to next_path and return
+        end
       end
 
-      redirect_to action: :edit
+      edit
+      render :edit
     end
 
     private
 
     def form_name
-      "document_type_upload_form"
+      self.class.form_class.name.underscore
     end
 
     def self.form_class
-      DocumentTypeUploadForm
+      if DocumentNavigation::REQUIRED_DOCUMENT_TYPES.include?(document_type)
+        RequiredDocumentUploadForm
+      else
+        DocumentTypeUploadForm
+      end
```

Comment:
This is one of the less than ideal consequences of created a child class for the RequiredDocumentUploadForm. Creating a different controller for required document upload forms would be an option or just combining the logic into the DocumentTypeUploadForm which was the approach that @bengolder suggested in an original spike that we did with @jenny-heath.